connectivity/deploy/manifests/node-local-dns-ds.yaml

```diff
@@ -0,0 +1,82 @@
+apiVersion: apps/v1
+kind: DaemonSet
```

Comment:
In which cases will this test be run?

The background of this question is that users can run the CLI connectivity tests in production environments, and I wouldn't want to deploy another DNS DaemonSet in a live environment by default as this could potentially cause at least confusion, if not active disruption to the cluster. Even if this doesn't cause disruption, deploying a DaemonSet to thousands of nodes in a production environment could take significant resources.

This might be a case where we should move the node-local DNS setup pieces into a dedicated GitHub workflow so that this test more closely matches the way that our users will configure and test this feature.

Reply:
(I see code that should make this test only run when LRP is enabled, but still, deploying a node-local DNS DaemonSet as part of the test is breaking the assumption "treat the cluster like a live production environment".)